server/cluster.go

```diff
@@ -448,18 +448,38 @@ func (c *RaftCluster) SetStoreWeight(storeID uint64, leader, region float64) err
 }
 
 func (c *RaftCluster) checkStores() {
+	var offlineStores []*metapb.Store
+	var upStoreCount uint64
+
 	cluster := c.cachedCluster
-	for _, store := range cluster.getMetaStores() {
+
+	for _, store := range cluster.GetStores() {
 		if store.GetState() != metapb.StoreState_Offline {
-			continue
+			if store.GetState() == metapb.StoreState_Up && !store.IsLowSpace(cluster.GetLowSpaceRatio()) {
+				upStoreCount++
+			}
 		}
-		if c.storeIsEmpty(store.GetId()) {
-			err := c.BuryStore(store.GetId(), false)
+		offlineStore := store.Store
+		// If the store is empty, it can be buried.
+		if cluster.isPrepared() && cluster.getStoreRegionCount(offlineStore.GetId()) == 0 {
```

Comment:
Do we really need to check `isPrepared()` here? I've noticed that after #1225 all regions will be put to `RegionsInfo` on startup (regardless of whether the leader is nil)